### PR TITLE
Backend/feat/#674/update beckn api

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Implemented beckn update API for POSTPAID payment. In case if BAP verify payment, it will send payment completed update to BPP 

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by DonDokidon — APPROVED** [review]

File: Backend/lib/beckn-spec/src/Beckn/Types/Core/Taxi/OnUpdate/OnUpdateEvent/RideCompletedEvent.hs [hunk 1]
```diff
@@ -32,7 +33,8 @@ data RideCompletedEvent = RideCompletedEvent
   { id :: Text,
     update_target :: Text,
     quote :: RideCompletedQuote,
-    fulfillment :: FulfillmentInfo
+    fulfillment :: FulfillmentInfo,
+    payment :: Payment
```
Comment (hkmangla) [inline Backend/lib/beckn-spec/src/Beckn/Types/Core/Taxi/OnUpdate/OnUpdateEvent/RideCompletedEvent.hs:37]: Can we make it backward compatible?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs [hunk 2]
```diff
@@ -149,12 +153,20 @@ buildOnUpdateMessage req@RideCompletedBuildReq {} = do
       OnUpdate.RideCompleted
         RideCompletedOU.RideCompletedEvent
           { id = ride.bookingId.getId,
-            update_target = "fulfillment.state.code,quote.price,quote.breakup",
+            update_target = "fulfillment.state.code,quote.price,quote.breakup,payment.uri",
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs:156]: Can you make it backward compatible?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs (hunk 2 above)
Comment (roman-bodavskiy) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs:156]: @hkmangla content o this field never used on bap side
So these changes backward compatible itself

File: Backend/lib/beckn-spec/src/Beckn/Types/Core/Taxi/OnUpdate/OnUpdateEvent/RideCompletedEvent.hs (hunk 1 above)
Comment (roman-bodavskiy) [reply Backend/lib/beckn-spec/src/Beckn/Types/Core/Taxi/OnUpdate/OnUpdateEvent/RideCompletedEvent.hs]: Fixed

**Review by hkmangla — APPROVED** [review]

**Pull request merged**